core/src/pow/types.rs

```diff
@@ -318,10 +318,10 @@ impl ProofOfWork {
 ///
 /// The hash of the `Proof` is the hash of its packed nonces when serializing
 /// them at their exact bit size. The resulting bit sequence is padded to be
-/// byte-aligned.
-/// We form a PROOFSIZE*edge_bits integer by by packing the PROOFSIZE edge indices together, 
-/// pad it with up to 7 0 bits to a multiple of 8 bits, write as a little endian byte array, 
-/// and hash with blake2b with 256 bit digest.
+/// byte-aligned. We form a PROOFSIZE*edge_bits integer with edge index i occupying 
+/// bits i * edge_bits through (i+1) * edge_bits - 1, by packing the PROOFSIZE edge
```

Comment:
It reads better in this order:

We form a PROOFSIZE*edge_bits integer by packing the PROOFSIZE edge indices together, with edge index i occupying bits i * edge_bits through (i+1) * edge_bits - 1,